### PR TITLE
4500zenja1/ File amends

### PR DESCRIPTION
**Pull request by 4500zenja1**

### CODE REVIEW

File: Elevel/Elevel.Api/Controllers/FileController.cs [hunk 1]
```diff
@@ -19,17 +19,17 @@ public FileController(IFileService fileService)
...
         /// </summary>
         /// <param name="formFiles">Files (as a list)</param>
         /// <returns></returns>
         [HttpPost(nameof(Upload))]
         public IActionResult Upload([Required] List<IFormFile> formFiles)
         {
-            _fileService.UploadFiles(formFiles);
-            return Ok();
+            var pathFile = _fileService.UploadFiles(formFiles);
+            return Ok(new { pathfile = pathFile.Result });
```
Comment (DShcharbakou) [inline Elevel/Elevel.Api/Controllers/FileController.cs:32]: delete this .Result

File: Elevel/Elevel.Infrastructure.Services/Implementation/FileService.cs [hunk 2]
```diff
@@ -3,31 +3,36 @@
...
-                if (file.Length <= 0) return;
-                var fileExtension = Path.GetExtension(file.FileName);
-                var filePath = Path.Combine(folderPath, $"myFile{DateTime.Now:yyyy_MM_dd-HH_mm_ss}{fileExtension}");
-                using (var stream = new FileStream(filePath, FileMode.Create))
-                {
-                    await file.CopyToAsync(stream);
-                }
-            });
+            var file = files[0];
+            if (file.Length <= 0) return "";
```
Comment (DShcharbakou) [inline Elevel/Elevel.Infrastructure.Services/Implementation/FileService.cs:29]: more readable 
if()
{
  return string.Empty;
}

File: Elevel/Elevel.Api/Controllers/FileController.cs [hunk 3]
```diff
@@ -40,10 +41,13 @@ public IActionResult Upload([Required] List<IFormFile> formFiles)
...
-        public IActionResult Download([Required] string filePath)
+        public FileResult Download([Required] string filePath)
         {
-            var file =  _fileService.DownloadFile(filePath);
-            return File(file.Content, file.ContentType, file.FileName);
+            var file = _fileService.DownloadFile(filePath);
+            return new FileStreamResult(file.Stream, file.ContentType)
+            {
+                FileDownloadName = file.FileName,
+            };
```
Comment (DShcharbakou) [inline Elevel/Elevel.Api/Controllers/FileController.cs:50]: FileDownloadName = _fileService.DownloadFile(filePath).FileName,

File: Elevel/Elevel.Api/Controllers/FileController.cs [hunk 4]
```diff
@@ -19,17 +20,17 @@ public FileController(IFileService fileService)
...
         /// <param name="formFiles">Files (as a list)</param>
         /// <returns></returns>
         [HttpPost(nameof(Upload))]
-        public IActionResult Upload([Required] List<IFormFile> formFiles)
+        public async Task<IActionResult> Upload([Required] List<IFormFile> formFiles)
         {
-            _fileService.UploadFiles(formFiles);
-            return Ok();
+            var pathFile = await _fileService.UploadFiles(formFiles);
+            return Ok(new { pathfile = pathFile });
```
Comment (DShcharbakou) [inline Elevel/Elevel.Api/Controllers/FileController.cs:33]: return Ok(new { pathfile = await _fileService.UploadFiles(formFiles)});

File: Elevel/Elevel.Infrastructure.Services/Implementation/FileService.cs [hunk 5]
```diff
@@ -3,41 +3,48 @@
...
-            var contentType = "APPLICATION/octet-stream";
+            FileStream fs = new(Path.Combine(folderPath, fileName), FileMode.Open, FileAccess.Read);
+            var contentType = "application/octet-stream";
             return new FileType
             {
-                Content = content,
+                Stream = fs,
                 ContentType = contentType,
                 FileName = fileName
             };
```
Comment (DShcharbakou) [inline Elevel/Elevel.Infrastructure.Services/Implementation/FileService.cs:50]: Stream = new(Path.Combine(folderPath, fileName), FileMode.Open, FileAccess.Read),
ContentType = "application/octet-stream",

**Review by DShcharbakou — DISMISSED** [review]

**Review by DShcharbakou — APPROVED** [review]

File: Elevel/Elevel.Api/Controllers/FileController.cs (hunk 3 above)
Comment (4500zenja1) [reply Elevel/Elevel.Api/Controllers/FileController.cs:49]: If that, we need to request DownloadFile at least one more time (for file.Stream and/or file.ContentType), and that would be not good for memory/time/unexpected errors, imho

**Review by DShcharbakou — APPROVED** [review]

**Review by parys — APPROVED** [review]

**Pull request merged**